### ISSUE TITLE
Code sign fails at native module

### ISSUE THREAD

**Issue by mldev-py** [electron-userland/electron-builder#4388]
Hi, @sethlu 
I use native module node_sqlite3.node for my packaged nodejs script and the nodejs script is packaged by pkg and used by electron application.
So I manually attach this node_sqlite3.node file into the resources folder of electron app and codesign fails with this node_sqlite3.node
I passed the codesign for the packaged nodejs executable by changing the extension but fails at node_sqlite3.node
Seems electron-osx-codesign fails with native module?
That is the main issue I'm getting.

```
/Users/lucasgonze/Documents/src/freemium-fork/orchard-data-tests/electron-app/build/mac/Viva Data.app/Contents/Resources/node_sqlite3.node: main executable failed strict validation
```

**Comment by sethlu** [comment]
Hi @mldev-py! Thanks for the question!
But before I can give advice, I'll need some clarifications 😸

From my understanding, your Electron app (specifically, the node scripts) depends on `node_sqlite3.node`.

You tried to manually attach the `node_sqlite3.node` in the app bundle's `Resources` folder but "codesign fails." By saying "codesign fails," do you mean that the app is first packaged, code signed with `electron-osx-sign`, and finally you manually moved in the `node_sqlite3.node` file? So the code sign verification fails.

By saying "changing the extension of the executable", do you mean changing the extension of your `AppName.app` or `node_sqlite3.node`? I'm not sure which extension this refers to?

**Comment by mldev-py** [comment]
Hi, @sethlu 
Sorry for my late response.
I had a separate nodejs script which needs to be used with ElectronJs application.
So I packaged this script with `pkg` library and attached `node_sqlite3.node` with the executable as `pkg` library requires that.
When I say `changing the extension of the executable` is that changing the extension of the packaged file of node.js script as package file packaged by `pkg` module has a problem with code signing.
But here, I had a problem with `node_sqlite3.node` code signing since I got `node_sqlite3.node: main executable failed strict validation` issue.
I've figured out this by integrating `node_sqlite3.node` module into the executable file with a trick and now the problem has been resolved.
Anyway, I appreciate your response.
Thank you
